### ISSUE TITLE
Add the ability to override some or all settings in zappa_settings.json with command line arguments

### ISSUE THREAD

**Issue by jordanambra** [Miserlou/Zappa#1112]
Generally speaking, it would be nice to be able to override values in zappa settings using command line arguments. This might not work for more complex settings, but at least the strings and numbers could be set.

## Context
For example, on my development machine, I have 7 or 8 different AWS profiles, so in my zappa_settings.json I have profile_name set for the development environment. However, I am also using CI to deploy, so my staging and production environments do NOT have profile_name set so that they can just use the defaults.

With those settings, if I want to manually run a command against staging or production from my computer, I have to go edit zappa_setting.json and add the profile_name, run the command, then remove it again so I don't break the build.

## Expected Behavior
It would be nice to be able to run a command like this:

`zappa tail staging --profile_name some_profile_name`

## Implementation

If you agree with this direction, I can probably implement it.

**Comment by mcrowson** [comment]
Why not just have different stages with these settings?

**Comment by mcrowson** [comment]
would setting the environment variable for AWS_PROFILE work for this?

**Comment by jordanambra** [comment]
For some situations it would theoretically work, but AWS_PROFILE appears to be overridden by what's in zappa_settings.json or is ignored during Zappa commands.

`AWS_PROFILE=obviousgarbage zappa tail production` doesn't override what's in zappa settings, or fail at all like you might expect. Is it being used? I don't think it is.

Anyway, that's just one example parameter. I'd like others as well.

**Comment by Miserlou** [comment]
I'm not sure I quite understand the reason for this. Surely each of your environments should be a different stage in your settings file?

**Comment by jordanambra** [comment]
Yes, each environment contains its own stage. The place I run commands from changes though.

For example, from CI the profile is default, whereas from my development machine it's (companyname). If I wanted to run a tail against staging I can't, because the profile name in the settings file is blank so that CI can deploy.

If you are developing on a team, people could potentially have different names for AWS profiles as well (default vs a specified profile name again, most likely).

**Comment by mcrowson** [comment]
Aside from this profile setting, are there others that would need to be overridden?

**Comment by Miserlou** [comment]
We could maaaaybe do something like

```bash
$ ZAPPA_PROFILE_NAME=local ZAPPA_LAMBDA_SIZE=128 zappa deploy dev
```

Where everything after `ZAPPA_` get lowercased and treated as a priority setting for that stage, but I don't think it's very elegant. Having custom `--kwargs` for every desired setting seems ever worse though..

**Comment by mcrowson** [comment]
The workflow where anyone, from any environment would want to call `zappa update` just seems like a strange workflow to me in general and I suspect would lead to other strange issues. Wouldn't you want a consistent CI service to test/build/deploy for you?

**Comment by jordanambra** [comment]
I don't want to call deploy or update, I want to call tail in particular. The whole reason for asking this is that I DO want a consistent CI service and I don't want to be hacking settings files if I happen to want to perform one of the commands that should not be done via CI, like zappa certify, zappa tail, or zappa invoke as a one-time customization or monitoring.

As far as Zappa and just general good programming and CI practices goes, I'm doing everything properly. My environments are properly defined and isolated. I promise I'm not doing anything crazy.

Any settings that would be overridden would be the ones that aren't baked into the environment itself necessarily and could conceivably be run outside of CI or by different members of a development team. Looking through the command list and settings file options, these come to mind:

- zappa certify staging --certificate /path/to/secret/local/cert --certificate_key /secret/local/key --certificate-chain /secret/local/chain/key
- zappa rollback production --profile-name myclientname
- zappa tail staging --profile-name myclientname
- zappa status production --profile-name myclientname
- zappa invoke staging 'myapp.some_function' --include "some_special_library_just_for_now"

Is that more helpful to have these examples?

**Comment by Miserlou** [comment]
Jordan, would the envvar based solution I mentioned previously suffice?

**Comment by jordanambra** [comment]
Yeah. ENV vars or arguments, either one.
